src/prototype_room_creepbuilder.js

```diff
@@ -211,7 +211,14 @@ Room.prototype.getPartsStringDatas = function(parts, energyAvailable) {
 Room.prototype.getSettings = function(creep) {
   const role = creep.role;
   const updateSettings = roles[role].updateSettings && roles[role].updateSettings(this, creep);
-  const settings = _.merge(roles[role].settings, updateSettings);
+  if (config.debug.spawn) {
+    this.log(`getSettings updateSettings: ${JSON.stringify(updateSettings)}`);
+    this.log(`getSettings roles[role].settings: ${JSON.stringify(roles[role].settings)}`);
+  }
+  const settings = _.merge({}, roles[role].settings, updateSettings);
```

Comment:
Wow, nice catch, tricky to find, thx

Reply:
Yeah took me some time to find it but in the end I found it :-)